Objects/tupleobject.c

```diff
@@ -1036,8 +1039,14 @@ tupleiter_len(PyObject *self, PyObject *Py_UNUSED(ignored))
 {
     _PyTupleIterObject *it = _PyTupleIterObject_CAST(self);
     Py_ssize_t len = 0;
+#ifdef Py_GIL_DISABLED
+    Py_ssize_t idx = FT_ATOMIC_LOAD_SSIZE_RELAXED(it->it_index);
+    if (it->it_seq && idx >= 0 && idx < PyTuple_GET_SIZE(it->it_seq))
```

Comment:
```suggestion
    if (idx < PyTuple_GET_SIZE(it->it_seq))
```
?

The  `it->it_seq` is not set to null any longer and idx cannot be negative. (I do not think `it_index` can overflow through the increments in `tupleiter_next`)